CONTRIBUTING.md

```diff
@@ -1,8 +1,78 @@
 # JFrog welcomes community contribution!
 
-Before we can accept your contribution, process your GitHub pull requests, and thank you full heartedly, we request that you will fill out and submit JFrog's Contributor License Agreement (CLA).
+Before we can accept your contribution, process your GitHub pull requests, and thank you full-heartedly, we request that you will fill out and submit JFrog's Contributor License Agreement (CLA).
 
 [Click here](https://secure.echosign.com/public/hostedForm?formid=5IYKLZ2RXB543N) to submit the JFrog CLA.
 This should only take a minute to complete and is a one-time process.
 
 *Thanks for Your Contribution to the Community!* :-)
+
+## Pull Request Process ##
+
+- Fork this repository.
+- Clone the forked repository to your local machine and perform the proposed changes. You can use [artifactory-user-plugins-devenv](https://github.com/JFrogDev/artifactory-user-plugins-devenv) to setup your development environment.
+- Make sure your changes follow the Acceptance Criteria below.
+- When you are confident about the changes, create a Pull Request pointing to the master branch of this repository. 
+
+## Acceptance Criteria ##
+
+Plugins must adhere to a number of conventions, mostly in the way the plugin files are laid out. This allows tools such as the `artifactory-user-plugins-devenv` and our internal testing system to properly recognize and work with plugins. These conventions are as follows:
+
+- Plugin files must be placed inside a plugin folder. The plugin groovy file and its folder must share the same name. The plugin folder can be placed inside a category folder to group plugins under the same subject.
+- If the plugin depends on an external file that it accesses while running, such as a configuration file, this file must have the same name as the plugin groovy file.
+- Update the `README.md` file to be consistent with the changes. If you are proposing a new plugin or if the `README.md` file is missing, use the template below to create a new one.
+- Provide at least one automated test case. If you cannot provide working test cases, use the template below to add an always-fail test case to your plugin. This will make it easier for us to identify plugins that lack test cases and need our attention.
```

Comment:
Preface this with "Ideally provide at least one automated test case" and go on from there, and then with "If you cannot..., PLEASE use ..."